### ISSUE TITLE
[QUESTION] Double-titration of sea salt alkalinity?

### ISSUE THREAD

**Issue by hazel008** [geoschem/geos-chem#1506]
### Ask a question about GEOS-Chem:
This is Hansen Cao from Prof. Mat Evans' group at York University. We are working on halogen heterogeneous chemistry on sea salt. I found the following lines in sulfate_mod.F in v13.4:

```F90
          ! We also need to update the SALAAL and SALCAL species here
          ! when we are using KPP to perform sulfur chemistry, otherwise
          ! SALAAL and SALCAL will not get set properly.  Think about
          ! a better way to abstract this code later.
          !   -- Bob Yantosca (09 Sep 2021)
          IF ( .not. State_Chm%Do_SulfateMod_SeaSalt ) THEN
             Spc(I,J,L,id_SALAAL) = AlkA
             Spc(I,J,L,id_SALCAL) = AlkC
          ENDIF
```
Given that in fullchem.eqn in KPP, we already have the sea salt chemistry titration reactions as beblow, I am very confused by the alkalinity update in sulfate_mod.F
```kconfig
SO2  + SALAAL + O3  = SO4 - SALAAL       : K_MT(1);
HCl  + SALAAL       = SALACL             : K_MT(2);
HNO3 + SALAAL       = NIT                : K_MT(3);
SO2  + SALCAL + O3  = SO4s - SALCAL      : K_MT(4);
HCl  + SALCAL       = SALCCL             : K_MT(5);
HNO3 + SALCAL       = NITs               : K_MT(6);
```
I wonder what do you mean by "We also need to update the SALAAL and SALCAL species here when we are using KPP to perform sulfur chemistry, otherwise SALAAL and SALCAL will not get set properly"?

I am asking this because Becky and I found v13.4 might double count the alkalinity titration through alkalinity updates in both seasalt_mod and KPP.

Thanks,
Hansen

**Comment by yantosca** [comment]
Thanks for writing @hazel008.  I am looping in @beckyalexander and @msl3v who have been looking into this issue.

I believe that I added the comment there so that the SALACL and SALCCL species wouldn't be zero if you were using KPP.  I think this is due to some legacy code.  AlkA and AlkC has been computed in ISORROPIA, and are stored in State_Chm.  But this may no longer be necessary since we recently added SALCAL and SALCCL species to keep track of the alkalinity.

I will defer to @beckyalexander and @msl3v as they have a better grasp of the issue than I do.

**Comment by yantosca** [comment]
Also tagging @mje1398

**Comment by sdeastham** [comment]
This is an interesting problem. 1 molec/cm3 is an astonishingly low concentration, so I'm curious to know how much chemistry is expected for concentrations of 2.0e-4 molec/cm3. If the chemistry is non-trivial then it may be necessary to have more complex checks in place here - lower the threshold below 1 molec/cm3 definitely caused problems for us as discussed in #1115.

**Comment by yantosca** [comment]
I would caution against putting in a complex check (i.e. IF/ELSE) since that is a known performance hit.  Better to make a parallel routine with a different threshold that is only called for the affected reactions.

**Comment by beckyalexander** [comment]
We definitely have a bug in the calculation of sea salt alkalinity titration. Sea salt alkalinity is titrated twice, once in sulfate_mod and again in KPP. This explains why sea salt alkalinity is so much lower in the latest version of GEOS-Chem. I commented out the updates to sea salt alkalinity in sulfate_mod, but then KPP died. I figured out that there was an issue with the way that the rate constant was being calculated for the titration of sea salt alkalinity for the uptake of SO2+O3 on alkaline sea salt. I fixed this and the model runs successfully and the sea salt alkalinity looks like it did in older versions of the code. I will submit a git patch soon.

**Comment by beckyalexander** [comment]
Attached is a patch file that gets rid of the sea salt alkalinity titration in sulfate_mod when it is computed in KPP to avoid double counting the alkalinity titration. There are also some fixes to the units used for the sea salt alkalinity tracers in KPP and the calculation of the third-order rate constants for SO2 + O3 + SALAAL and SO2 + O3 + SALCAL reactions in KPP.
[alk-patch-file.txt](https://github.com/geoschem/geos-chem/files/10036618/alk-patch-file.txt)
[SALAAL_SALCAL.pdf](https://github.com/geoschem/geos-chem/files/10036621/SALAAL_SALCAL.pdf)

**Comment by beckyalexander** [comment]
I forgot to mention that I also changed the initial cloud pH from 0.0 to 4.5.

**Comment by yantosca** [comment]
Thanks Becky.  I'll create a PR from your patch file and test.

**Comment by yantosca** [comment]
@beckyalexander: what is the parent commit of your patch?  You said you were using 13.3.4?

**Comment by yantosca** [comment]
@beckyalexander: Nevermind, I figured it out.

**Comment by msulprizio** [comment]
The fix for this has now been merged into the `dev/14.1.0` branch.
